src/System.IO.FileSystem/src/System/IO/PathHelpers.Windows.cs

```diff
@@ -40,9 +40,16 @@ internal static string GetFullPathInternal(string path)
             if (path == null)
                 throw new ArgumentNullException("path");
 
-            string pathTrimmed = path.TrimStart(TrimStartChars).TrimEnd(TrimEndChars);
-
-            return Path.GetFullPath(Path.IsPathRooted(pathTrimmed) ? pathTrimmed : path);
+            if (PathInternal.IsExtended(path))
+            {
+                // Don't want to trim extended paths
+                return Path.GetFullPath(path);
+            }
+            else
+            {
+                string pathTrimmed = path.TrimStart(TrimStartChars).TrimEnd(TrimEndChars);
```

Comment:
At some point (doesn't need to be now), it'd be nice to consolidate the start and end trimming into a single operation, so that if both need to trim, we create a single string rather than two.